rasa/utils/tensorflow/rasa_layers.py

```diff
@@ -810,6 +811,15 @@ def _get_transformer_dimensions(
         if isinstance(transformer_units, dict):
             transformer_units = transformer_units[attribute]
         if transformer_layers > 0 and (not transformer_units or transformer_units < 1):
+            rasa.shared.utils.io.raise_warning(
+                f"`{TRANSFORMER_SIZE}` is set to "
+                f"`{transformer_layers}` for "
+                f"{attribute}, but a positive size is required when using "
+                f"`{NUM_TRANSFORMER_LAYERS} > 0`. {attribute} will proceed, using "
+                f"`{TRANSFORMER_SIZE}={DEFAULT_TRANSFORMER_SIZE}`. "
+                f"Alternatively, specify a different value in the component's config.",
+                category=UserWarning,
+            )
```

Comment:
```suggestion
                f"`{TRANSFORMER_SIZE}` is set to "
                f"`{transformer_units}`, "
                f"but a positive size is required when using "
                f"`{NUM_TRANSFORMER_LAYERS} > 0`. We will proceed, using "
                f"`{TRANSFORMER_SIZE}={DEFAULT_TRANSFORMER_SIZE}`. "
                f"Alternatively, specify a different value in the component's config.",
                category=UserWarning,
            )
```
This way it's a bit non-specific but still understandable.